openstates/md/bills.py

```diff
@@ -165,10 +170,132 @@ def parse_bill_votes(self, doc, bill):
                 vote.add_source(href)
                 yield vote
 
+    def parse_bill_votes_new(self, doc, bill):
+        elems = doc.xpath('//a')
+
+        # MD has a habit of listing votes twice
+        seen_votes = set()
+
+        for elem in elems:
+            href = elem.get('href')
+            if (href and "votes" in href and href.endswith('pdf') and
+                    ("Senate" in href or "House" in href) and href not in seen_votes):
+                seen_votes.add(href)
+                vote = self.parse_vote_pdf(href, bill)
+                vote.add_source(href)
+                yield vote
+
+    def parse_vote_pdf(self, vote_url, bill):
+
+        filename, response = self.urlretrieve(vote_url)
+
+        text = convert_pdf(filename, type='text').decode()
+        lines = text.splitlines()
+
+        if 'senate' in vote_url:
+            chamber = 'upper'
+        else:
+            chamber = 'lower'
+
+        date_string = lines[0].split('Date:')[1].strip()
+        date = datetime.datetime.strptime(date_string, "%b %d, %Y %I:%M (%p)")
+
+        pageIndex = -1
+        for index, line in enumerate(lines):
+            if 'Yeas' in line and 'Nays' in line:
+                pageIndex = index
+                break
+
+        yes_count = -1
```

Comment:
Why do these counters start at -1? If this is necessary, please add a comment to clarify.

Reply:
I prefered -1, because if some case occurs with a bad pdf design, it will throw an error as yes_count in model only allows value 0 or positive.

Reply:
You shouldn't rely on the model to catch bad vote counts that way; you should do it yourself (eg, `if yes_count == 0 and no_count == 0 …` or `if not pageIndex …` then throw an error).

Reply:
Actually I am relying here on the validator, which works in ```--scrape``` time, but after hearing you, I think checking it here only is the best idea.